pkg/testutil/testutil.go

```diff
@@ -17,15 +17,33 @@ package testutil
 import (
 	"net/url"
 	"runtime"
+	"strings"
 	"testing"
+	"time"
 )
 
-// WARNING: This is a hack.
-// Remove this when we are able to block/check the status of the go-routines.
-func ForceGosched() {
-	// possibility enough to sched up to 10 go routines.
-	for i := 0; i < 10000; i++ {
-		runtime.Gosched()
+func runnableGoroutineNumber() int {
```

Comment:
comment added. It refers the net pkg test function: https://github.com/golang/go/blob/b016eba489820b9091e4f39610a84f697d8eb0f9/src/net/http/main_test.go#L26

I have browsed through the runtime library and find no luck. Hopefully go will expose the state of goroutines explicitly.